test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,225 @@
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
+
+    def load_config(self, cfg):
+        """
+        Create a hook configuration file with the provided contents.
+        """
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=5,
+                           starttime=self.server.ctime)
+        self.logger.info("Current config: %s" % cfg)
+        # Restart MoM to work around PP-993
+        self.mom.restart()
+
+    @timeout(600)
+    def test_cgroups_race_condition(self):
+        """
+        Test to ensure a cgroups event does not read the cgroups file system
+        while another event is writing to it. By submitting 1000 instant jobs,
+        the events should collide at least once.
+        """
+        pcpus = 0
+        with open('/proc/cpuinfo', 'r') as desc:
+            for line in desc:
+                if re.match('^processor', line):
+                    pcpus += 1
+        if pcpus < 8:
+            self.skipTest("Test requires at least 8 physical CPUs")
+
+        attr = {'job_history_enable': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+        self.load_config(self.cfg0)
+        now = time.time()
+        j = Job(TEST_USER, attrs={ATTR_J: '0-1000'})
+        j.create_script(self.true_script)
+        jid = self.server.submit(j)
+        jid = jid.split('[')
+        done = False
+        for i in range(0, 1000):
+            sjid = jid[0] + "[" + str(i) + "]"
+            if done:
+                self.server.expect(
+                    JOB, {'job_state': 'F'}, id=sjid, extend='x')
+            else:
+                try:
+                    self.server.expect(
+                        JOB, {'job_state': 'X'}, id=sjid, extend='tx')
+                except PtlExpectError:
+                    done = True
+                    self.server.expect(
+                        JOB, {'job_state': 'F'}, id=sjid, extend='x')
+
+            if i % 100 == 0:
+                self.mom.log_match(msg="IOError", starttime=now,
+                                   existence=False, max_attempts=1, n="ALL")
+                now = time.time()
```

Comment:
remove resetting of the time to make sure to catch previous errors which you might have missed due to race conditions.

Reply:
Removed